src/components/NewCommentForm.tsx

```diff
@@ -1,68 +1,167 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Post } from '../types/Post';
+import { CommentData } from '../types/Comment';
+
+type Props = {
+  post: Post;
+  addComment: (v: CommentData & { postId: number }) => Promise<void>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ post, addComment }) => {
+  const [name, setName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [hasEmailError, setHasEmailError] = useState(false);
+
+  const [body, setBody] = useState('');
+  const [hasBodyError, setHasBodyError] = useState(false);
+
+  const [isSending, setIsSending] = useState(false);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setHasNameError(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setHasEmailError(false);
+  };
+
+  const handleBodyChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setBody(e.target.value);
+    setHasBodyError(false);
+  };
+
+  const reset = () => {
+    setName('');
+    setHasNameError(false);
+
+    setEmail('');
+    setHasEmailError(false);
+
+    setBody('');
+    setHasBodyError(false);
+  };
+
+  const submit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    setIsSending(true);
+
+    const nameError = !name.trim();
+    const emailError = !email.trim();
+    const bodyError = !body.trim();
+
+    setHasNameError(nameError);
+    setHasEmailError(emailError);
+    setHasBodyError(bodyError);
+
+    if (nameError || emailError || bodyError) {
+      setIsSending(false);
+
+      return;
+    }
+
+    const newComment = {
+      name,
+      body,
+      email,
+      postId: post.id,
+    };
+
+    addComment(newComment).finally(() => setIsSending(false));
```

Comment:
Consider handling errors in the `addComment` promise. Currently, errors from the `addComment` function are not caught, which might lead to unhandled promise rejections.